crates/lunatic-common-api/src/lib.rs

```diff
@@ -11,6 +11,27 @@ pub fn get_memory<T>(caller: &mut Caller<T>) -> Result<Memory> {
         .or_trap("Export `memory` is not a memory")
 }
 
+// Call guest to allocate a Vec of size `size`
+pub fn allocate_guest_memory<'a, T: Send>(
```

Comment:
Maybe we should use a universal code for "out of memory" errors? For calls that will allocate guest memory we could then require the guest to send an opaque pointer to which we write the error in case the process ran out of memory. On the other hand, maybe we should just let it crash? Depends on whether the process developer wants to have a graceful shutdown or not in such a case. I would personally like to crash the process and try to figure out what data I've been reading that caused memory to run out.

Reply:
Trapping should suffice for now I guess. It's either an untrusted program trying to use more memory than it should or a memory limit was not properly set, which means it could be a bug or a miscalculation. Either way, crashing and trapping is probably the most straightforward option.